modules/tutorials/pages/java-tutorial/add-course-enrollment.adoc

```diff
@@ -0,0 +1,143 @@
+
+        QueryOptions courseQueryOptions = QueryOptions.queryOptions();
+        courseQueryOptions.parameters(JsonObject.create().put("courseName", course));
+        courseQueryOptions.scanConsistency(QueryScanConsistency.REQUEST_PLUS);
+
+        final QueryResult result = cluster.query("select META().id, crc.* " +
+                        "from `student-bucket`.`art-school-scope`.`course-record-collection` crc " +
+                        "where crc.`course-name` = $courseName", courseQueryOptions);
+
+        return result.rowsAsObject().get(0);
+
+    }
+}
+----
++
+NOTE: Because this is a tutorial, you do not need to add an error check to make sure that your collection has returned an item.
+In a live application, though, error checks must be made to prevent errors and keep the application running.
++
+<1> Calls `Cluster.connect` to create a channel to the named server and supplies your username and password to authenticate the connection.
+<2> Retrieves the bucket you set up when you xref:tutorial-install-server.adoc[installed the Couchbase server].
+<3> Forces the application to wait until the bucket is ready.
+<4> Retrieves the `art-school-scope` collection from the scope.
+<5> These three methods retrieve Hilary's student record, the `graphic design` course record, and the `art history` course record.
+Each method uses a {sqlpp} call to retrieve a single record from each collection.
+<6> Couchbase does not have a native date type, so the common practice is to store dates as strings.
+<7> Stores the `enrollments` inside the student record as an array.
+`JsonArray.create()` creates an empty list structure.
+<8> Adds two JSON elements to the `enrollments` array: the course that the enrollment relates to, and the date that the student enrolled in the course. 
+To avoid repeating data all over the database, you store a reference to the course instead of the entire course record itself in this field.
+This means that you do not have to search through every single record if the course details change, 
+<9> Adds the `enrollments` array to Hilary's student record.
+<10> Commits the changes to the collection.
+The `upsert` function call takes the key of the record you want to insert or update and the record itself as parameters.
+If the `upsert` call finds a document with a matching ID in the collection, it updates the document. 
+If there is no matching ID, it creates a new document.
++
+. Open a terminal window and navigate to your `student` directory.
+. Run the command `mvn install` to pull in all the dependencies and rebuild your application.
+. Run the following command to insert the student record into the collection:
++
+[source, sh]
+----
+mvn exec:java -Dexec.mainClass="AddEnrollments" -Dexec.cleanupDaemonThreads=false
+----
++
+. Go to your Couchbase database in your browser and check the `student-record-collection` for the new information you just added to Hilary's student record.
```

Comment:
Is Hilary with one l or two? I swear I thought it was two in the previous file lol